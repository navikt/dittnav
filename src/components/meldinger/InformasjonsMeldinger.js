import React, { Component } from 'react';
import { node, bool } from 'prop-types';
import { FormattedMessage as F } from 'react-intl';
import PanelMedIkon from '../common/PanelMedIkon';
import PanelOverskrift from '../common/PanelOverskrift';
import IkonBeskjed from '../../assets/IkonBeskjed';

class InformasjonsMeldinger extends Component {
  render() {
    const children = [];

    if (this.props.visGenerellInfo) {
      const generellInfoOverskrift = <PanelOverskrift overskrift={this.props.generellInfo} type="Normaltekst" />;

      children.push(// eslint-disable-line function-paren-newline
        <PanelMedIkon
          className="beskjed"
          overskrift={generellInfoOverskrift}
          key="generell"
          erInformasjonsmelding
        >
          <IkonBeskjed />
        </PanelMedIkon>,
      );
    }

    if (this.props.isMeldeKortUser && this.props.visMeldekortbrukerInfo) {
      const meldekortOverskrift = <PanelOverskrift overskrift={this.props.meldekortOverskrift} type="Normaltekst" />;

      children.push(// eslint-disable-line function-paren-newline
        <PanelMedIkon
          className="beskjed"
          overskrift={meldekortOverskrift}
          key="melderkort"
          erInformasjonsmelding
        >
          <IkonBeskjed />
        </PanelMedIkon>,
      );
    }
    return (<>{children}</>);
  }
}

InformasjonsMeldinger.propTypes = {
  visGenerellInfo: bool,
  visMeldekortbrukerInfo: bool,
  isMeldeKortUser: bool,
  meldekortOverskrift: node,
  generellInfo: node,
};

InformasjonsMeldinger.defaultProps = {
  visGenerellInfo: false,
  visMeldekortbrukerInfo: false,
  isMeldeKortUser: false,
  meldekortOverskrift: <F id="meldekortbruker.informasjonsmelding" />,
  generellInfo: <F id="generell.informasjonsmelding" />,
};

export default InformasjonsMeldinger;
