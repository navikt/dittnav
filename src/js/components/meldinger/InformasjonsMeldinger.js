import React, { Component } from 'react';
import PropTypes from 'prop-types';

class InformasjonsMeldinger extends Component {
  render() {
    if (!this.props.infoMessages) return null;
    const children = [];
    if (this.props.infoMessages['generell.informasjonsmelding']) {
      children.push(// eslint-disable-line function-paren-newline
        <div className="message" key="general">
          <span className="icon default-icon" aria-label="alarm-ikon" />
          <div className="texts">
            <p>{this.props.infoMessages['generell.informasjonsmelding']}</p>
          </div>
        </div>,
      );
    }
    if (this.props.isMeldeKortUser && this.props.infoMessages['meldekortbruker.informasjonsmelding']) {
      children.push(// eslint-disable-line
        <div className="message" key="melderkort">
          <span className="icon meldekort-icon" aria-label="alarm-ikon" />
          <div className="texts">
            <p>{this.props.infoMessages['meldekortbruker.informasjonsmelding']}</p>
          </div>
        </div>,
      );
    }
    return (<React.Fragment>{children}</React.Fragment>);
  }
}

export const InfoMessagesType = PropTypes.shape({
  'generell.informasjonsmelding': PropTypes.string,
  'meldekortbruker.informasjonsmelding': PropTypes.string,
});

InformasjonsMeldinger.propTypes = {
  infoMessages: InfoMessagesType,
  isMeldeKortUser: PropTypes.bool,
};

InformasjonsMeldinger.defaultProps = {
  infoMessages: null,
  isMeldeKortUser: false,
};

export default InformasjonsMeldinger;
