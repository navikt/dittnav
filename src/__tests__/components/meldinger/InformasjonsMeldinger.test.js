import * as React from 'react';
const ReactTestRenderer = require('react-test-renderer');
import InformasjonsMeldinger from 'js/components/meldinger/InformasjonsMeldinger'
import wrapIntl from 'js/Intl';

test('InformasjonsMeldinger without messages', () => {
  const component = ReactTestRenderer.create(wrapIntl(<InformasjonsMeldinger />));
  expect(component.toJSON()).toMatchSnapshot();
});

test('InformasjonsMeldinger with general', () => {
  const infoMessages = {'generell.informasjonsmelding': 'Hello'};
  const component = ReactTestRenderer.create(wrapIntl(<InformasjonsMeldinger infoMessages={infoMessages} />));
  expect(component.toJSON()).toMatchSnapshot();
});

test('InformasjonsMeldinger with meldekort', () => {
  const infoMessages = {'meldekortbruker.informasjonsmelding': 'Melderkort message'};
  const component = ReactTestRenderer.create(wrapIntl(<InformasjonsMeldinger infoMessages={infoMessages} isMeldeKortUser={true} />));
  expect(component.toJSON()).toMatchSnapshot();
});

test('InformasjonsMeldinger with meldekort but not user', () => {
  const infoMessages = {'meldekortbruker.informasjonsmelding': 'Melderkort message'};
  const component = ReactTestRenderer.create(wrapIntl(<InformasjonsMeldinger infoMessages={infoMessages} isMeldeKortUser={false} />));
  expect(component.toJSON()).toMatchSnapshot();
});

test('InformasjonsMeldinger with both', () => {
  const infoMessages = {'meldekortbruker.informasjonsmelding': 'Melderkort message', 'generell.informasjonsmelding': 'Hello'};
  const component = ReactTestRenderer.create(wrapIntl(<InformasjonsMeldinger infoMessages={infoMessages} isMeldeKortUser={true} />));
  expect(component.toJSON()).toMatchSnapshot();
});
