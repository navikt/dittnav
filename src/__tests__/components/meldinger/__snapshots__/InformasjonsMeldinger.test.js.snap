// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InformasjonsMeldinger with both 1`] = `
Array [
  <div
    className="message"
>
    <span
        aria-label="alarm-ikon"
        className="icon default-icon"
    />
    <div
        className="texts"
    >
        <p>
            Hello
        </p>
    </div>
</div>,
  <div
    className="message"
>
    <span
        aria-label="alarm-ikon"
        className="icon meldekort-icon"
    />
    <div
        className="texts"
    >
        <p>
            Melderkort message
        </p>
    </div>
</div>,
]
`;

exports[`InformasjonsMeldinger with general 1`] = `
<div
  className="message"
>
  <span
    aria-label="alarm-ikon"
    className="icon default-icon"
  />
  <div
    className="texts"
  >
    <p>
      Hello
    </p>
  </div>
</div>
`;

exports[`InformasjonsMeldinger with meldekort 1`] = `
<div
  className="message"
>
  <span
    aria-label="alarm-ikon"
    className="icon meldekort-icon"
  />
  <div
    className="texts"
  >
    <p>
      Melderkort message
    </p>
  </div>
</div>
`;

exports[`InformasjonsMeldinger with meldekort but not user 1`] = `null`;

exports[`InformasjonsMeldinger without messages 1`] = `null`;
